src/client.rs

```diff
@@ -103,23 +117,293 @@ impl<'a> Client<'a> {
         subject_key: &str,
         subject_attributes: &SubjectAttributes,
     ) -> Result<Option<AssignmentValue>> {
+        self.get_assignment_inner(flag_key, subject_key, subject_attributes, None, |x| x)
+    }
+
+    /// Retrieves the assignment value for a given feature flag and subject.
+    ///
+    /// If the subject is not eligible for any allocation, returns `Ok(None)`.
+    ///
+    /// If the configuration has not been fetched yet, returns `Ok(None)`.
+    /// You should call [`Client::start_poller_thread`] before any call to
+    /// `get_string_assignment()`. Otherwise, the client will always return `None`.
+    ///
+    /// It is recommended to wait for the Eppo configuration to get fetched with
+    /// [`PollerThread::wait_for_configuration()`].
+    ///
+    /// # Errors
+    ///
+    /// Returns an error in the following cases:
+    /// - [`Error::FlagNotFound`] if the requested flag configuration was not found.
+    /// - [`Error::InvalidType`] if the requested flag has an invalid type or type conversion fails.
+    /// - [`Error::ConfigurationParseError`] or [`Error::ConfigurationError`] if the configuration
+    /// received from the server is invalid.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # fn test(client: &eppo::Client) {
+    /// let assignment = client
+    ///     .get_string_assignment("a-string-flag", "user-id", &[
+    ///         ("language".into(), "en".into())
+    ///     ].into_iter().collect())
+    ///     .unwrap_or_default()
+    ///     .unwrap_or("default_value".to_owned());
+    /// # }
+    /// ```
+    pub fn get_string_assignment(
+        &self,
+        flag_key: &str,
+        subject_key: &str,
+        subject_attributes: &SubjectAttributes,
```

Comment:
Please add a `default_value: String` parameter, or whatever the type is returned, to all public `get_*_assignment` methods. This will match our APIs in other SDKs: https://github.com/Eppo-exp/js-client-sdk-common/blob/main/src/client/eppo-client.ts#L45-L78